hal/riscv-rvv/src/features2d/fast.cpp

```diff
@@ -0,0 +1,289 @@
+#include "rvv_hal.hpp"
+#include "common.hpp"
+#include <cfloat>
+
+namespace cv { namespace rvv_hal { namespace features2d {
+
+using RVV_VECTOR_TYPE = vuint8m4_t;
+
+
+// Since uint16_t range is 0 to 65535, row stride should be less than 65535/6 = 10922
+inline void makeOffsets(int16_t pixel[], vuint16m2_t& v_offset, int64_t row_stride, int patternSize)
+{
+    uint16_t pixel_u[25];
+
+    switch(patternSize) {
+    case 16:
+        // set min element (pixel[9] = -1 + row_stride * -3) as the base addr
+        pixel_u[0] = 1 + row_stride * 6;
+        pixel_u[1] = 2 + row_stride * 6;
+        pixel_u[2] = 3 + row_stride * 5;
+        pixel_u[3] = 4 + row_stride * 4;
+        pixel_u[4] = 4 + row_stride * 3;
+        pixel_u[5] = 4 + row_stride * 2;
+        pixel_u[6] = 3 + row_stride * 1;
+        pixel_u[7] = 2 + row_stride * 0;
+        pixel_u[8] = 1 + row_stride * 0;
+        pixel_u[9] = 0 + row_stride * 0;
+        pixel_u[10] = -1 + row_stride * 1;
+        pixel_u[11] = -2 + row_stride * 2;
+        pixel_u[12] = -2 + row_stride * 3;
+        pixel_u[13] = -2 + row_stride * 4;
+        pixel_u[14] = -1 + row_stride * 5;
+        pixel_u[15] = 0 + row_stride * 6;
+
+        for (int i = 16; i < 25; i++)
+        {
+            pixel_u[i] = pixel_u[i - 16];
+        }
+        v_offset = __riscv_vle16_v_u16m2(pixel_u, 25);
```

Comment:
CI node has 128x2-bit width of register, so max number of uint16_t elements in a lane is 16. 25 is just wrong.

Reply:
Fixed.